tests/test_odl.py

```diff
@@ -1925,7 +1932,493 @@ def canonicalize_author_name(self, identifier, working_display_name):
+        """
+        return self._expires
+
+
+class TestLicenseInfo:
+    """Represents information about the current state of a license stored in the License Info Document."""
+
+    def __init__(
+        self, remaining_checkouts: int, available_concurrent_checkouts: int
+    ) -> None:
+        """Initialize a new instance of TestLicenseInfo class.
+
+        :param remaining_checkouts: Total number of checkouts left for a License
+        :param available_concurrent_checkouts: Number of concurrent checkouts currently available
+        """
+        self._remaining_checkouts: int = remaining_checkouts
+        self._available_concurrent_checkouts: int = available_concurrent_checkouts
+
+    @property
+    def remaining_checkouts(self) -> int:
+        """Return the total number of checkouts left for a License.
+
+        :return: Total number of checkouts left for a License
+        """
+        return self._remaining_checkouts
+
+    @property
+    def available_concurrent_checkouts(self) -> int:
+        """Return the number of concurrent checkouts currently available.
+
+        :return: Number of concurrent checkouts currently available
+        """
+        return self._available_concurrent_checkouts
+
+    def __str__(self) -> str:
+        """Return a JSON representation of a part of the License Info Document."""
+        return json.dumps(
+            {
+                "checkouts": {
+                    "left": self.remaining_checkouts,
+                    "available": self.available_concurrent_checkouts,
+                }
+            }
+        )
+
+
+class TestODLExpiredItemsReaper(DatabaseTest, BaseODLTest):
+    """Base class for all ODL reaper tests."""
+
+    ODL_PROTOCOL = ODLAPI.NAME
+    ODL_TEMPLATE_DIR = os.path.join(BaseODLTest.base_path, "files", "odl")
+    ODL_TEMPLATE_FILENAME = "feed_template.xml.jinja"
+    ODL_REAPER_CLASS = ODLExpiredItemsReaper
+
+    def _create_importer(self, collection, http_get):
+        """Create a new ODL importer with the specified parameters.
+
+        :param collection: Collection object
+        :param http_get: Use this method to make an HTTP GET request.
+            This can be replaced with a stub method for testing purposes.
+
+        :return: ODLImporter object
+        """
+        importer = ODLImporter(
+            self._db,
+            collection=collection,
+            http_get=http_get,
+        )
+
+        return importer
+
+    def _get_test_feed(self, licenses: List[TestLicense]) -> str:
+        """Get the test ODL feed with specific licensing information.
+
+        :param licenses: List of ODL licenses
 
+        :return: Test ODL feed
+        """
+        env = Environment(
+            loader=FileSystemLoader(self.ODL_TEMPLATE_DIR), autoescape=select_autoescape()
+        )
+        template = env.get_template(self.ODL_TEMPLATE_FILENAME)
+        feed = template.render(licenses=licenses)
+
+        return feed
```

Comment:
This uses a Jinja2 template to generate an ODL feed for whatever licenses you want to test. This way you can create the License objects for your use case and generate the feed, without having to pregenerate a separate feed for each use case.

This isn't how we've done this kind of testing up to this point, but it does make it easier to test a large number of  slight variant files.